extensions/azurecore/src/azureResource/services/tenantService.ts

```diff
@@ -2,29 +2,17 @@
  *  Copyright (c) Microsoft Corporation. All rights reserved.
  *  Licensed under the Source EULA. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
-
-import * as request from 'request';
+import { SubscriptionClient } from '@azure/arm-subscriptions';
 
 import { azureResource } from '../azure-resource';
 import { IAzureResourceTenantService } from '../interfaces';
+import { Account } from 'azdata';
 
 export class AzureResourceTenantService implements IAzureResourceTenantService {
-	public async getTenantId(subscription: azureResource.AzureResourceSubscription): Promise<string> {
-		const requestPromisified = new Promise<string>((resolve, reject) => {
-			const url = `https://management.azure.com/subscriptions/${subscription.id}?api-version=2014-04-01`;
-			request(url, function (error, response, body) {
-				if (response.statusCode === 401) {
-					const tenantIdRegEx = /authorization_uri="https:\/\/login\.windows\.net\/([0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12})"/;
-					const teantIdString = response.headers['www-authenticate'];
-					if (tenantIdRegEx.test(teantIdString)) {
-						resolve(tenantIdRegEx.exec(teantIdString)[1]);
-					} else {
-						reject();
-					}
-				}
-			});
-		});
+	public async getTenantId(subscription: azureResource.AzureResourceSubscription, account: Account, credentials: any): Promise<string> {
```

Comment:
Oh interesting, I wonder if this is going to help with the issues we've been seeing with querying azure resources. Did you do this upgrade for any reasoning along that or just because it's better to do it with an actual library?

Reply:
It just seemed weird that we were using the library for subscriptions but not for tenants.